### ISSUE TITLE
Known-good compatible IDEs?

### ISSUE THREAD

**Issue by vemv** [gfredericks/debug-repl#9]
Hey there,

project looking nice. It looks better than other alternatives (including a similiar lib I quickly drafted).

It'd be helpful to know, what editors/IDEs is this known to work well with?

In particular, CIDER interests me. Can't check compat myself because I'm stuck at cider 0.16 while current version is 0.19. And `debug-repl` is not working for me with 0.16 - but that might be a corner case, better not to bother you yet, in principle.

tldr, it would be nice to add `Known-good compatible IDEs` to the README.

Cheers - Victor

**Comment by gfredericks** [comment]
I personally use it mostly with cider, and I believe I was using 0.16 for a while.

Did you try the version that was released a couple days ago?

One limitation is that it's an nrepl middleware, so if the IDE isn't running an nrepl repl, then it can't work.

**Comment by vemv** [comment]
Thanks for the reply, it's reassuring to know that this can work in Emacs :)

I could isolate the issue I'm having. For reproducing it, the following profile:

```clojure
 :user {:dependencies [[com.gfredericks/debug-repl "0.0.10"]
        :repl-options {:nrepl-middleware [com.gfredericks.debug-repl/wrap-debug-repl]}
        :plugins [[refactor-nrepl "2.4.0"  :exclusions [org.clojure/tools.logging]]]}
```

...will trigger:

```
No implementation of method: :send of protocol: #'clojure.tools.nrepl.transport/Transport found for class: com.gfredericks.debug_repl$syncronous_new_session$reify__2682
```

**Comment by gfredericks** [comment]
I suspect this comes up when one piece of your tooling is using an older version of nrepl where the namespace is named `clojure.tools.nrepl.*` and another is using a newer version where the namespace is named `nrepl.*`

debug-repl uses the newer if it's on the classpath. the error references the older

If possible, I'd recommend figuring out what's putting the old version of nrepl on the classpath and upgrading that or suppressing that dependency somehow

**Comment by vemv** [comment]
Thanks for the diagnostic!

As per my experimentation, `[com.gfredericks/debug-repl "0.0.10"]` will:

* work if no `nrepl` depency is present at all
* work if nrepl is 0.3.1
* not work for nrepl 0.4.0, 0.4.4, 0.6.0

Did reproduce excluding first all junk from my profiles.clj. Perhaps the compat code could be refined?

---

Btw, now that I could get debug-repl to lauch, I can confirm that it works like a charm in CIDER :) really seamless repl integration.

**Comment by gfredericks** [comment]
for testing newer versions of nrepl, I would focus on the question "is `org.clojure/tools.nrepl` on my classpath, and if so, why?

Figure out how to get it off the classpath.

If it's not on the classpath at all, then this must be an unrelated issue.

**Comment by vemv** [comment]
> for testing newer versions of nrepl, I would focus on the question "is `org.clojure/tools.nrepl` on my classpath, and if so, why?"

Good focus!

Given that I could reproduce the issue even with a minimal

```
{:user {:dependencies [[com.gfredericks/debug-repl "0.0.10"]
                       [nrepl "0.6.0"]]
        :repl-options {:nrepl-middleware [com.gfredericks.debug-repl/wrap-debug-repl]}}}
```

, then the answer had to be "Leiningen". I updated from 2.8.1 to 2.9.0 and voila.

**Comment by gfredericks** [comment]
I added a [section to the README](https://github.com/gfredericks/debug-repl#nrepl-compatibility) explaining this kind of situation; do you think that's sufficient?

**Comment by vemv** [comment]
LGTM :)
